src/Players/util.cpp

```diff
@@ -1,15 +1,51 @@
 #include "hex-ai/Players/util.hpp"
 #include <iostream>
+#include <limits>
 
 using Action = GameState::HexState::Action;
 
 void Players::get_action(Action &a) {
```

Comment:
Yes, this is great. The main reason for the redundant documentation was for me to understand what was going on (not familiar with C++ --- learning while writing). 

I agree that the board size macro should be used wherever its applicable and contributes to clarity. 

And don't worry, you're not stifling my creativity - my next step would have been to attempt to de-duplicate just as you did, except to a lesser degree (since there are a few constructs you used that I don't yet recognize).